### PR TITLE
Development merge to main

### PR DESCRIPTION
**Pull request by HuiSunNCSU**
I request merge development to main branch for final demo.

### CODE REVIEW

**Comment by tingxiang-ji** [comment]
approve

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
+
```
Comment (HuiSunNCSU) [inline .github/workflows/ci.yml:10]: here is in diff hunks comment

File: .github/workflows/ci.yml
Comment (HuiSunNCSU) [inline .github/workflows/ci.yml:1]: here is the in file comment

File: backend/controllers/resume_controller.js [hunk 2]
```diff
@@ -83,6 +83,79 @@ exports.getResume = async (req, res) => {
...
+  try {
+    const user = await User.findOne({ _id: userId });
+
+    if (!user) {
+      console.log("User not found for userId:", userId);
+      return res.status(404).send({ error: "User not found" });
+    }
+
+    if (user.videoUrl) {
+      console.log(`Existing video URL found: ${user.videoUrl}`);
```
Comment (HuiSunNCSU) [inline backend/controllers/resume_controller.js:114]: 123

**Pull request merged**